lib/wallet/src/Cardano/Wallet/Primitive/Types/RewardAccount.hs

```diff
@@ -36,22 +36,26 @@ import Quiet
 
 -- | A reward account is used in group-type addresses for delegation.
 --
--- It is the public key of the account address.
+-- It is either the public key or script hash.
 --
-newtype RewardAccount = RewardAccount { unRewardAccount :: ByteString }
+data RewardAccount
+    = FromKeyHash ByteString
+    | FromScriptHash ByteString
     deriving (Generic, Eq, Ord)
     deriving (Show, Read) via (Quiet RewardAccount)
 
 instance NFData RewardAccount
 
 instance Buildable RewardAccount where
-    build = build . Hash @"RewardAccount" . unRewardAccount
+    build (FromKeyHash bs) = build . Hash @"RewardAccount" $ bs
+    build (FromScriptHash bs) = build . Hash @"RewardAccount" $ bs
 
 instance ToText RewardAccount where
-    toText = toText . Hash @"RewardAccount" . unRewardAccount
+    toText (FromKeyHash bs) = toText . Hash @"RewardAccount" $ bs
+    toText (FromScriptHash bs) = toText . Hash @"RewardAccount" $ bs
 
 instance FromText RewardAccount where
-    fromText = fmap (RewardAccount . getHash @"RewardAccount") . fromText
+    fromText = fmap (FromKeyHash . getHash @"RewardAccount") . fromText
```

Comment:
🤔 This wouldn't roundtrip... Could this be problematic for the DB?

If this is indeed a problem, maybe there's a middle ground between having `RewardAccount = FromKeyHash ByteString | FromScriptHash ByteString` everywhere and having two `newRewardBalanceFetcher` — e.g. having the same `RewardAccount` as before but letting `newRewardBalanceFetcher` operate on `Either RewardAccount RewardAccount` (or something isomorphic) as opposed to having two of them?